plugins/woocommerce/client/blocks/assets/js/atomic/blocks/product-elements/shared/use-is-descendent-of-single-product-template.tsx

```diff
@@ -1,14 +1,18 @@
 /**
  * External dependencies
  */
+import { store as editorStore } from '@wordpress/editor';
 import { useSelect } from '@wordpress/data';
+import { isString } from '@woocommerce/types';
 
 export const useIsDescendentOfSingleProductTemplate = () => {
 	const isDescendentOfSingleProductTemplate = useSelect( ( select ) => {
-		const store = select( 'core/edit-site' );
-		const postId = store?.getEditedPostId< string | undefined >();
+		// @ts-expect-error getEditedPostSlug is not typed
+		const postSlug = select( editorStore ).getEditedPostSlug();
 
-		return Boolean( postId?.includes( '//single-product' ) );
+		return isString( postSlug )
+			? postSlug.includes( 'single-product' )
```

Comment:
`getEditedPostSlug` returns just what's after `//` so for example `single-product` instead of `woocommerce//single-product`.